shared/libraries/opcuatms/opcuatms_client/src/objects/tms_client_signal_impl.cpp

```diff
@@ -161,4 +161,23 @@ StringPtr TmsClientSignalImpl::onGetRemoteId() const
     return String(deviceSignalId);
 }
 
+ErrCode TmsClientSignalImpl::getValue(IBaseObject** value)
+{
+    *value = nullptr;
+
+    // if descriptorNode exists, value node also exists
+    if (descriptorNodeId)
+    {
+        const auto valueNodeId = clientContext->getReferenceBrowser()->getChildNodeId(nodeId, "Value");
+        OpcUaVariant opcUaVariant = client->readValue(*valueNodeId);
+        if (!opcUaVariant.isNull())
+        {
+            BaseObjectPtr valuePtr = VariantConverter<IBaseObject, BaseObjectPtr>::ToDaqObject(opcUaVariant);
+            *value = valuePtr.addRefAndReturn();
+        }
```

Comment:
We could have a failsafe here to try and read the analogValue node if the value fails. It might be that a device only sends that one.

Also, it could be the case that the AnalogValue node exists, but not the Value one. In that case we should always read AnalogValue.